src/data/DataStore.ts

```diff
@@ -991,7 +1002,12 @@ class DataStore {
         // First frame use the last data.
         newIndices[sampledIndex++] = this.getRawIndex(len - 1);
         target._count = sampledIndex;
-        target._indices = newIndices;
+
+        const _newIndices = new (getIndicesCtor(this._rawCount))(sampledIndex);
+        for (let idx = 0; idx < sampledIndex; idx++) {
+            _newIndices[idx] = newIndices[idx];
+        }
+        target._indices = _newIndices;
```

Comment:
It's not required to create a new array which match lenght precisely.

Instead of `_indices.length`, we use `_count` to indicate the length of indices. So we can avoid creating dynamic arrays that may has worse performance than a static array.

In the code, we don't need to create a `_newIndices`. Just use the `newIndices`, it's fine even if it has empty items.

Reply:
got it.